pkg/operator/controller/ingress/deployment.go

```diff
@@ -221,6 +225,22 @@ func determineDeploymentReplicas(ic *operatorv1.IngressController, ingressConfig
 	return DetermineReplicas(ingressConfig, infraConfig)
 }
 
+func headerValues(values []operatorv1.IngressControllerHTTPHeader) string {
```

Comment:
See https://github.com/openshift/cluster-ingress-operator/blob/a29464eda9614ac3d70c185b572f4839195e7306/pkg/operator/controller/ingress/deployment.go#L1716.  I think you should be doing the escaping/sanitizing for HAProxy in the ingress controller before it gets to HAProxy, like the compression feature does for mime types.

Unit Testing for it: https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/deployment_test.go#L1780 

E2E Testing for it: https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/router_compression_test.go

Reply:
> See
> 
> https://github.com/openshift/cluster-ingress-operator/blob/a29464eda9614ac3d70c185b572f4839195e7306/pkg/operator/controller/ingress/deployment.go#L1716
> 
> . I think you should be doing the escaping/sanitizing for HAProxy in the ingress controller before it gets to HAProxy, like the compression feature does for mime types.

Router code is the best place to add sanitization logic as we have to fetch fields(header values) from route API as as well.

> Unit Testing for it: https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/operator/controller/ingress/deployment_test.go#L1780
> 
> E2E Testing for it: https://github.com/openshift/cluster-ingress-operator/blob/master/test/e2e/router_compression_test.go